src/feature/flow_chart_panel/views/AppGalleryModal.tsx

```diff
@@ -18,11 +18,12 @@ import {
   SelectItem,
   SelectTrigger,
 } from "@/components/ui/select";
-import { DialogClose } from "@radix-ui/react-dialog";
+import { useFlowChartState } from "@src/hooks/useFlowChartState";
 
 const subjectKeyList = ["fundamentals", "AI", "IO", "DSP"];
 const ignoreDir = [".github", "MANIFEST"];
 export const AppGalleryModal = () => {
+  const { isGalleryOpen, setIsGalleryOpen } = useFlowChartState();
```

Comment:
There is no reason for this to be global state, you can easily make this a `useState` and pass it down as a prop from `FlowChartTabView` instead.

Reply:
But I also use that inside the ```AppGalleryElement.tsx``` file to close down the model when we click an example

Reply:
You can pass it down through `AppGalleryLayout`'s props and then pass it down again to `AppGalleryElement`.